pkg/controller/sriovnetworknodepolicy/sriovnetworknodepolicy_controller.go

```diff
@@ -755,10 +755,8 @@ func (r *ReconcileSriovNetworkNodePolicy) syncDaemonSet(cr *sriovnetworkv1.Sriov
 		}
 	} else {
 		logger.Info("DaemonSet already exists, updating")
-		if ds.Spec.Template.Spec.Affinity != nil && in.Spec.Template.Spec.Affinity != nil {
-			if !reflect.DeepEqual(*ds.Spec.Template.Spec.Affinity.NodeAffinity, *in.Spec.Template.Spec.Affinity.NodeAffinity) {
-				ds.Spec.Template.Spec.Affinity.NodeAffinity = in.Spec.Template.Spec.Affinity.NodeAffinity
-			}
+		if !apiequality.Semantic.DeepEqual(ds.Spec, in.Spec) {
```

Comment:
Why need to compare the difference, can we just do the `client.Update` and let the api to figure out whether it should do the real update or not?

Reply:
Good point. Fixed.